worker/buildbot_worker/pb.py

```diff
@@ -157,22 +143,30 @@ def activity(self, res=None):
         pass
 
     def stopFactory(self):
-        ReconnectingPBClientFactory.stopFactory(self)
         self.stopTimers()
+        AutoLoginPBFactory.stopFactory(self)
 
 
 class Worker(WorkerBase, service.MultiService):
+    """The service class to be instantiated from buildbot.tac
+
+    to just pass a connection description string, set buildmaster_host and
+    port to None, and use conndescr.
+
+    maxdelay is deprecated if favour of using twisteds backoffPolicy.
+    """
     Bot = BotPb
 
-    def __init__(self, buildmaster_host, port, name, passwd, basedir,
+    def __init__(self,  buildmaster_host, port, name, passwd, basedir,
                  keepalive, usePTY=None, keepaliveTimeout=None, umask=None,
-                 maxdelay=300, numcpus=None, unicode_encoding=None,
-                 allow_shutdown=None, maxRetries=None):
```

Comment:
why remove maxRetries?

Reply:
Ope that must be a merge misstake. I'll fix it later tonight.